# Tests for 7-base_geometry

1. Import test
>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
>>>

2. Calculate area
>>> bg = BaseGeometry()
>>> bg.area()
Traceback (most recent call last):
    ...
Exception: area() is not implemented

3. Pass a non integer argument
>>> bg.integer_validator("name", "5")
Traceback (most recent call last):
    ...
TypeError: name must be an integer

4. Pass a negative argument
>>> bg.integer_validator("name", -5)
Traceback (most recent call last):
    ...
ValueError: name must be greater than 0

5. Pass a zero
>>> bg.integer_validator("name", -5)
Traceback (most recent call last):
    ...
ValueError: name must be greater than 0

6. No inputs for integer_validator
>>> bg.integer_validator()
Traceback (most recent call last):
    ...
TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

7. One input for integer_validator
>>> bg.integer_validator('name')
Traceback (most recent call last):
    ...
TypeError: integer_validator() missing 1 required positional argument: 'value'

8. Excess inputs to input_validator
>>> bg.integer_validator('name', 3, '23')
Traceback (most recent call last):
    ...
TypeError: integer_validator() takes 3 positional arguments but 4 were given

9. Passing excess arguments to area method
>>> bg.area(5)
Traceback (most recent call last):
    ...
TypeError: area() takes 1 positional argument but 2 were given

10. Passing a tuple to integer_validator
>>> bg.integer_validator('name', (3, '23'))
Traceback (most recent call last):
    ...
TypeError: name must be an integer

10. Passing a list to integer_validator
>>> bg.integer_validator('name', [3, '23'])
Traceback (most recent call last):
    ...
TypeError: name must be an integer

11. Passing a dict to integer_validator
>>> bg.integer_validator('name', {"age": '23'})
Traceback (most recent call last):
    ...
TypeError: name must be an integer

12. Passing a tuple to integer_validator
>>> bg.integer_validator('name', None)
Traceback (most recent call last):
    ...
TypeError: name must be an integer